### ISSUE TITLE
Occasional DNS error on mempool.space lookup

### ISSUE THREAD

**Issue by ok300** [breez/breez-sdk#530]
Sometimes getting a DNS error in the SDK:

```
[2023-10-17 22:47:04.134 ERROR breez_sdk_core::swap:301] failed to refresh swap status for address ... error sending request for url (https://mempool.space/api/address/.../txs): error trying to connect: dns error: task 86 was cancelled
```

When this happens, it happens for several minutes, even if I restart the SDK.

While the error was happening, I checked and the OS could ping and resolve mempool.space both on IPv4 and IPv6.

**Comment by ok300** [comment]
Searching for the exact error message in GH discussions, looks like it happens a lot:

https://github.com/search?q=%22error+trying+to+connect%3A+dns+error%22&type=discussions

**Comment by ok300** [comment]
Fixed by #708
